indico/modules/logs/controllers.py

```diff
@@ -210,18 +213,43 @@ def _process(self):
             raise BadRequest('Invalid log entry type')
         elif self.entry.data.get('attachments'):
             raise BadRequest('Cannot re-send email with attachments')
-        email = make_email(
-            to_list=self.entry.data['to'],
-            cc_list=self.entry.data.get('cc', []),
-            bcc_list=self.entry.data.get('bcc', []),
-            sender_address=self.entry.data['from'],
-            reply_address=self.entry.data.get('reply_to', []),
-            subject=self.entry.data['subject'],
-            body=self.entry.data['body'],
-            html=self.entry.data['content_type'] == 'text/html',
-            alternatives=self.entry.data.get('alternatives'),
+        elif self.entry.data.get('state') == 'pending':
+            raise BadRequest('Cannot re-send pending emails')
+
+        is_html_email = self.entry.data['content_type'] == 'text/html'
+        original_subject = self.entry.data['subject']
+        form = ResendEmailPrefaceForm(subject=original_subject, is_html_email=is_html_email)
+        if form.validate_on_submit():
+            email = self._build_email(form.data, is_html_email)
+            send_email(email, event=self.event, module=self.entry.module, user=self.entry.user,
+                       log_metadata=self.entry.meta, log_summary=f'Resent email: {original_subject}')
+            flash(_('The email has been re-sent.'), 'success')
+            return jsonify_data()
+        return jsonify_template('logs/resend_email_dialog.html', form=form, entry=self.entry)
+
+    def _build_email(self, form_data, is_html_email):
+        email_data = self.entry.data
+        body = email_data['body']
+        alternatives = email_data.get('alternatives') or []
+        if preface := form_data['preface']:
+            if is_html_email:
+                body = f'{preface}{body}'
+                alternatives = [
+                    (f'{strip_tags(preface)}\n\n{alt_body}', mimetype)
```

Comment:
I'd rather do `html_to_markdown(preface, inline_links=False).strip()` like we do in reminders. That way formatting is not *completely* lost.